### ISSUE TITLE
Night master publish happens even when there are no changes, attempting to re-publish the previous version

### ISSUE THREAD

**Issue by jonthysell** [microsoft/react-native-windows#7547]
**TLDR:** The ADO pipeline runs the nightly scheduled publish whenever a branch has new commits, and since beachball bumps the version by pushing a `applying package updates ***NO_CI***` commit, tomorrow is added to the schedule even if no other commits come in. If no other commits come in, then the publish runs with the same version and binaries as yesterday, and only fails at the very last moment, when it tries to push the results to a NuGet feed.

Beachball bumps the versions if there's been changes and commits the `applying package updates ***NO_CI***` commit. If there's no actual changes by the next day, then when publish runs, beachball sees there's no changes and doesn't do anything, just outputting `Nothing to bump, skipping publish!`.

However, this doesn't stop the pipeline. So the publish continues with the same version number, wasting time building, and ultimately it's the release pipeline that complains when it tries to push new packages with the same version number.

The publish pipeline should stop if beachball stops the bump, or never even start if the latest commit is from a `applying package updates ***NO_CI***` commit.
